src/features/chat/Chat.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { useEffect, useState, useRef, KeyboardEvent } from 'react'
+// import { useHistory } from 'react-router'
+import cn from 'classnames'
+import routes from '../../common/constants/routes.json'
+import { useHistory } from 'react-router-dom'
+import { ChatItem, ChatItemProps } from './ChatItem'
+import {
+  ChatMessage,
+  ChatMessageProps,
+  DIR_RECV,
+  DIR_SENT,
+} from './ChatMessage'
+import editIcon from '../../common/assets/icons/ico-edit.svg'
+import chatMenuIcon from '../../common/assets/icons/ico-chat-menu.svg'
+import msgEmojiIcon from '../../common/assets/icons/ico-chatmsg-emoji.svg'
+import msgAttachIcon from '../../common/assets/icons/ico-chatmsg-link.svg'
+import msgSendIcon from '../../common/assets/icons/ico-chatmsg-send.svg'
+import chatCloseIcon from '../../common/assets/icons/ico-close2.svg'
+import styles from './Chat.module.css'
+import { mockChats, curUser } from './mock-data'
+
+export type LocationState = {
+  param: string
+}
+
+export default function Chat(): JSX.Element {
+  const [newMsgPlaceholder, setNewMsgPlaceholder] = useState(true)
+  const [newMsg, setNewMsg] = useState('')
+  const [activeChatId, setActiveChatId] = useState<number>(mockChats[0].id)
+  const [activeChat, setActiveChat] = useState<ChatItemProps>(mockChats[0])
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [chats, setChats] = useState<ChatItemProps[]>(mockChats) // DEBUG: init state
```

Comment:
During development you can just ignore unused variables, and before submitting PR please remove them instead of disabling eslint rule

Reply:
setChats is not used now but will be used with real chats data. I can ignore it, but check-eslint don't ignore it and block commit.